homeassistant/components/sharkiq/sharkiq.py

```diff
@@ -0,0 +1,283 @@
+        raise NotImplementedError()
+
+    @property
+    def is_online(self) -> bool:
+        """Tell us if the device is online."""
+        return self.coordinator.device_is_online(self.sharkiq.serial_number)
+
+    @property
+    def name(self) -> str:
+        """Device name."""
+        return self.sharkiq.name
+
+    @property
+    def serial_number(self) -> str:
+        """Vacuum API serial number (DSN)."""
+        return self.sharkiq.serial_number
+
+    @property
+    def model(self) -> str:
+        """Vacuum model number."""
+        if self.sharkiq.vac_model_number:
+            return self.sharkiq.vac_model_number
+        return self.sharkiq.oem_model_number
+
+    @property
+    def device_info(self) -> Dict:
+        """Device info dictionary."""
+        return {
+            "identifiers": {(DOMAIN, self.serial_number)},
+            "name": self.name,
+            "manufacturer": SHARK,
+            "model": self.model,
+            "sw_version": self.sharkiq.get_property_value(
+                Properties.ROBOT_FIRMWARE_VERSION
+            ),
+        }
+
+    @property
+    def supported_features(self) -> int:
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_SHARKIQ
+
+    @property
+    def is_docked(self) -> Optional[bool]:
+        """Is vacuum docked."""
+        return self.sharkiq.get_property_value(Properties.DOCKED_STATUS)
+
+    @property
+    def error_code(self) -> Optional[int]:
+        """Return the last observed error code (or None)."""
+        return self.sharkiq.error_code
+
+    @property
+    def error_message(self) -> Optional[str]:
+        """Return the last observed error message (or None)."""
+        if not self.error_code:
+            return None
+        return self.sharkiq.error_text
+
+    @property
+    def operating_mode(self) -> Optional[str]:
+        """Operating mode.."""
+        op_mode = self.sharkiq.get_property_value(Properties.OPERATING_MODE)
+        return OPERATING_STATE_MAP.get(op_mode)
+
+    @property
+    def recharging_to_resume(self) -> Optional[int]:
+        """Return True if vacuum set to recharge and resume cleaning."""
+        return self.sharkiq.get_property_value(Properties.RECHARGING_TO_RESUME)
+
+    @property
+    def state(self):
+        """
+        Get the current vacuum state.
+
+        NB: Currently, we do not return an error state because they can be very, very stale.
+        In the app, these are (usually) handled by showing the robot as stopped and sending the
+        user a notification.
+        """
+        if self.recharging_to_resume:
+            return STATE_RECHARGING_TO_RESUME
+        if self.is_docked:
+            return STATE_DOCKED
+        return self.operating_mode
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique id of the vacuum cleaner."""
+        return f"sharkiq-{self.serial_number:s}-vacuum"
```

Comment:
No need to add domain or type, it is always checked for uniqueness including those 2 values.
```suggestion
        return self.serial_number
```